src/internal/small_map.rs

```diff
@@ -146,6 +146,13 @@ impl<K, V> SmallMap<K, V> {
     }
 }
 
+impl<K: PartialEq + Eq + Hash, V> Index<&K> for SmallMap<K, V> {
+    type Output = V;
+    fn index(&self, key: &K) -> &V {
+        &self.get(key).unwrap()
```

Comment:
Reading `terms[package]` is nice but probably treacherous. I think it would be better if we could avoid this. Then the caller is aware of when they are calling `unwrap()`.